### ISSUE TITLE
turn off warning for UnmarshalXML, MarshalXML, MarshalXMLAttr, UnmarshalXMLAttr

### ISSUE THREAD

**Issue by plandem** [golang/lint#407]
It's implementation of standard xml interface, what kind of comments do you expect here? In most cases it will be something like:
```go
//Unmarshal XXX
//Marshal XXX
```

Do you really think that it will improve somehow quality of documentation?

E.g. there is no warning about **String()**

P.S.: right now, in some cases, we have to comment that stuff to 'pass rules' of quality code (e.g.: https://github.com/avelino/awesome-go/issues/2024)

**Comment by dominikh** [comment]
> Do you really think that it will improve somehow quality of documentation?

It is generally better not to assume stupidity or hostility. Instead of thinking that we somehow love unnecessary comments, you could assume the simpler and less offending option: nobody thought to add these methods to the list of exceptions, and nobody asked for them to be on the list yet, either.

> P.S.: right now, in some cases, we have to comment that stuff to 'pass rules' of quality code (e.g.: avelino/awesome-go#2024)

Surely golint cannot be made responsible for how people use it. In fact, the README specifically states that

> The suggestions made by golint are exactly that: suggestions. Golint is not perfect, and has both false positives and false negatives. Do not treat its output as a gold standard

**Comment by plandem** [comment]
Sorry, if it sounded offended - nothing like that. 

I created ticket because maybe it's good idea to remove such warnings. But the same time it was a question to get idea why they exists - maybe I missed something and they are here on purpose.

**Comment by mvdan** [comment]
Thank you for submitting this issue! As per https://github.com/golang/go/issues/38968, we are freezing and deprecating golint. There's no drop-in replacement to golint per se, but you should find that [Staticcheck](https://staticcheck.io/) works well in encouraging good Go code, much like golint did in the past, since it also includes style checks. There's always `gofmt` and `go vet` too, of course.

If you would like to contribute further, I'd encourage you to engage [Staticcheck's issue tracker](https://github.com/dominikh/go-tools/issues) or look at [vet's open issues](https://github.com/golang/go/issues?q=is%3Aissue+is%3Aopen+cmd%2Fvet+in%3Atitle), as they are both actively maintained. If you have an idea that doesn't fit into either of those tools, you could look at [other Go linters](https://github.com/golangci/awesome-go-linters), or write your own - these days it's fairly straightforward with [go/analysis](https://pkg.go.dev/golang.org/x/tools/go/analysis).

To help avoid confusion, I'm closing all issues before we freeze the repository. If you have any feedback, you can leave a comment on the proposal thread where it was decided to deprecate golint - though note that the proposal has been accepted for nearly a year. Thanks!
